functions/form-payment-reconciliation/index.js

```diff
@@ -0,0 +1,149 @@
+const { SSMClient, GetParameterCommand } = require('@aws-sdk/client-ssm')
+const { generateLinkToCloudwatch, postToSlack } = require('./src/utils')
+
+const AWS_REGION = 'ap-southeast-1'
+
+const PARAMETER_STORE_NAME = 'staging-cron-payment'
+
+const SUB_DOMAIN = process.env.ENV_SITE_SUBDOMAIN || 'staging.'
+
+/** TODO: migrate to separate file */
+function getSecrets() {
+  const awsSsmClient = new SSMClient({ region: AWS_REGION })
+  const input = {
+    Name: PARAMETER_STORE_NAME,
+    WithDecryption: true,
+  }
+  const command = new GetParameterCommand(input)
+  return awsSsmClient
+    .send(command)
+    .then((d) => d.Parameter.Value)
+    .then((d) => {
+      const dn = d.split('\n')
+      return dn.reduce((accum, line) => {
+        const [key, value] = line.split('=')
+        accum[key] = value
+        return accum
+      }, {})
```

Comment:
I've had plenty of issues with parsers in the past on the config, even if it was dev controlled. The top 3 offender issues are:
1. accidental empty lines
2. accidental whitespaces around the equal sign, or as prefix/suffix
3. legitimate equal sign(s) in the values of the `key=value` pairs

So I recommend having some simple "standard" protection against those, with something like this this:
```suggestion
      const keyPairRe = /^\s*([^\s=]+)\s*=\s*(.+)$/
      const dn = d.trim().split(/[\r\n]+/)
      return dn.reduce((accum, line) => {
        const matches = line.match(keyPairRe)
        if (matches) {
            const [_, key, value] = matches
            accum[key] = value.trim()
        }
        return accum
      }, {})
```

Reply:
fixed in [95ab1e8](https://github.com/opengovsg/FormSG/pull/6102/commits/95ab1e89b5031f60b29d3019a75651cfb24f589d)